### ISSUE TITLE
Add a check to see if training loss has gone to infinity

### ISSUE THREAD

**Issue by wrapperband** [karpathy/char-rnn#123]
I've had training loss go to infinity a few times now. 

I thought it may have been due to bad characters, but it has now happened on, tested data, so I assume sometimes the net goes unstable.

I have tried to recover those brains and they seem to remain unbstable, even if tightly trained. So the reason, possible improvements to handle infinity (drop neurons option? or net scan tool), or if those brains might be "creative"

An option to stop processing if learning rate goes to infinity would be a good first quick fix.
